wormhole-connect/src/utils/routes/cosmosGateway.ts

```diff
@@ -559,15 +567,19 @@ export class CosmosGatewayRoute extends BaseRoute {
     return CosmWasmClient.create(tmClient);
   }
 
-  isTransferCompleted(
+  async isTransferCompleted(
     destChain: ChainName | ChainId,
     message: SignedMessage,
   ): Promise<boolean> {
     if (!isSignedWormholeMessage(message))
       throw new Error('Signed message is not for gateway');
-    return isCosmWasmChain(destChain)
-      ? wh.isTransferCompleted(CHAIN_ID_WORMCHAIN, hexlify(message.vaa))
-      : new BridgeRoute().isTransferCompleted(destChain, message);
+
+    if (!isCosmWasmChain(destChain)) {
+      return new BridgeRoute().isTransferCompleted(destChain, message);
+    }
+
+    const destTx = await this.fetchRedeemedEvent(message);
+    return !!destTx;
```

Comment:
Is finding the redeem event the only way to tell if a gateway transaction has been completed?

Reply:
After the transfer is completed in wormchain, there's a time window until the IBC transfer to the destination cosmos chain completes. Until then, the user won't have the funds available in their wallet. I thought it'd be a "harder" criteria than just checking the redeem on wormchain.

Perhaps it's a bit much and the time window can be negligible. But some QA and user reports showed that it can happen.